### ISSUE TITLE
ACC [ui5-tree] Tree lacks support for aria-describedby

### ISSUE THREAD

**Issue by don-obrien** [SAP/ui5-webcomponents#5593]
### **Bug Description**
sf-tree web component does not have support for customizing/populating the aria-describedby attribute, as well as a default implementation of the property for the tree.  
We should not be using aria-label (e.g. accessible-name or accessible-name-ref) for description
https://www.w3.org/TR/WCAG20-TECHS/ARIA1.html

Need this functionality as soon as possible to address SF customer defects.

### **Expected Behavior**
consumers of the component expect to be able to customize/populate the aria-describedby attribute on a tree component via some property on the sf-tree , e.g. accessibileDescription
Would expect a default behavior of describing the structure (e.g. tree) and basic navigation (arrow keys, key(s) for selection, etc.)
users expect to hear descriptive text at the correct level of detail in a screen reader.

### **Steps to Reproduce**
1. Go to web component api/playground and/or code
1.1 (https://sap.github.io/ui5-webcomponents/playground/components/Tree/)
1.2. https://github.com/SAP/ui5-webcomponents/blob/a8a4acdd7e919fb6ef32a8587748b82441a6379c/packages/main/src/Tree.js
2. notice that there is no support for aria-description
3. notice that there is no default aria-description for the component

### **Isolated Example**
If possible, please provide an isolated example by forking [CodeSandbox](https://codesandbox.io/s/71r1x5o51q?fontsize=14&module=%2Findex.html).

### **Context**
- UI5 Web Components version: {1.4.0}
- OS/Platform: {N/A}
- Browser: {Chrome}
- Affected component: {sf-tree}

### **Log Output / Stack Trace / Screenshots**
```
N/A
```
### **Priority**
- [ ] Low
- [] Medium
- [x] High
- [ ] Very High

The priority indicates the severity of the issue. To set the appropriate priority consider the following criteria:
- Breaks entire application or system - High or Very High
- Accessibility issue - Medium or High
- Functional issue - Medium or High
- Visual issue - Low or Medium

**Note:** The priority might be re-evaluated by the issue processor.

### **Stakeholder Info** (if applicable)

- Organization: {SFSF UXR}
- Business impact: {Affects accessibility for customers - customer ticket reports this.}

**Comment by MapTo0** [comment]
Dear @SAP/ui5-webcomponents-topic-p ,

SSF team requires accessibility properties for the Tree component.
I guess this should be discussed on a PO level and compared to what we have in SAPUI5.

Regards,
Martin

**Comment by don-obrien** [comment]
Can we a label for accessibility to this ticket?

**Comment by dobrinyonkov** [comment]
Hello @don-obrien,

After discussion with colleagues in the accessibility area I can confirm that
there is currently a limitation in providing ID references for elements with shadow DOM.
You can check for more information regarding this in our accessibility documentation
https://sap.github.io/ui5-webcomponents/playground/advanced/accessibility/

For that reason, it is not currently possible to add support for accessible descriptions.
We plan to expose an "accessibleDescription" property similar to the "accessibleName" and
"accessibleNameRef", which will map to aria-description property
https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaDescription
This will be implemented once ARIA 1.2 is officially published and it is not in draft.

However, we could introduce the "accessibleName" property which will map to the "aria-label" attribute as it seems like a reasonable solution.

Best Regards,
Dobrin

**Comment by dobrinyonkov** [comment]
Internal reference: BGSOFUIPIRIN-5831

**Comment by ilhan007** [comment]
Hello @SAP/ui5-webcomponents-topic-p - setting the priority to High based on the latest stakeholders input and prioritization. Consider this request on some of your next takt planings. Thank you.

Best,
ilhan

**Comment by dobrinyonkov** [comment]
Hi @don-obrien,

This will be planned and implemented during the next sprint. Unfortunately, it is possible that this will not be delivered before September 15 and more likely end of Sep with the release of 1.8.0.

Kind Regards,
Dobrin

**Comment by jdichev** [comment]
Hello @don-obrien,

As communicated by @dobrinyonkov, this item was planned and included as high priority in the sprint.

Best regards,
Jordan
